hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/targettable/TargetBeanQuery.java

```diff
@@ -76,16 +71,20 @@ public TargetBeanQuery(final QueryDefinition definition, final Map<String, Objec
         super(definition, queryConfig, sortIds, sortStates);
 
         if (HawkbitCommonUtil.isNotNullOrEmpty(queryConfig)) {
-            status = (Collection<TargetUpdateStatus>) queryConfig.get(SPUIDefinitions.FILTER_BY_STATUS);
-            overdueState = (Boolean) queryConfig.get(SPUIDefinitions.FILTER_BY_OVERDUE_STATE);
-            targetTags = (String[]) queryConfig.get(SPUIDefinitions.FILTER_BY_TAG);
-            noTagClicked = (Boolean) queryConfig.get(SPUIDefinitions.FILTER_BY_NO_TAG);
-            distributionId = (Long) queryConfig.get(SPUIDefinitions.FILTER_BY_DISTRIBUTION);
-            searchText = (String) queryConfig.get(SPUIDefinitions.FILTER_BY_TEXT);
-            targetFilterQueryId = (Long) queryConfig.get(SPUIDefinitions.FILTER_BY_TARGET_FILTER_QUERY);
+            String searchText = (String) queryConfig.get(SPUIDefinitions.FILTER_BY_TEXT);
             if (!StringUtils.isEmpty(searchText)) {
                 searchText = String.format("%%%s%%", searchText);
             }
+
+            filter = new FilterParams(
```

Comment:
Potential NPE in case `queryConfig` is null or empty => `filter` will be null. I think before this couldn't happen due to null-checks on primitive types. I'd suggest do set an empty filter in `case queryConfig` is null or empty.